client/src/components/Widget/Widget.tsx

```diff
@@ -23,40 +23,40 @@ const Widget: FunctionComponent<WidgetProps> = ({ color, data }) => {
     <S.Widget color={color}>
       <S.WidgetHeader color={color}>
         <Flex justifySpaceBetween>
-          <time dateTime={timeStamp.day}>{timeStamp.day}</time>
-          <time dateTime={timeStamp.time}>{timeStamp.time}</time>
+          <time dateTime={timeStamp?.day}>{timeStamp?.day}</time>
+          <time dateTime={timeStamp?.time}>{timeStamp?.time}</time>
         </Flex>
       </S.WidgetHeader>
       <S.WidgetBody>
         <Flex>
           <S.WidgetDetails>
-            <S.WidgetTemp>{data.temperature}°</S.WidgetTemp>
+            <S.WidgetTemp>{data?.temperature}°</S.WidgetTemp>
             <S.WidgetDetailsItem color={color}>
-              <span>Real Feel</span> {data.feelsLikeTemp}°
+              <span>Real Feel</span> {data?.feelsLikeTemp}°
             </S.WidgetDetailsItem>
             <S.WidgetDetailsItem color={color}>
-              <span>Wind</span> {data.windDirString}, speed {data.windSpeed}&nbsp;m/s, gust{' '}
-              {data.windGust} m/s
+              <span>Wind</span> {data?.windDirString}, speed {data?.windSpeed}&nbsp;m/s, gust{' '}
+              {data?.windGust} m/s
             </S.WidgetDetailsItem>
             <S.WidgetDetailsItem color={color}>
-              <span>Pressure</span> {data.pressure}
+              <span>Pressure</span> {data?.pressure}
             </S.WidgetDetailsItem>
             <S.WidgetDetailsItem color={color}>
-              <span>Humidity</span> {data.relHumidity}%
+              <span>Humidity</span> {data?.relHumidity}%
             </S.WidgetDetailsItem>
           </S.WidgetDetails>
           <S.WidgetDetails>
             <S.WidgetImg>
               <Flex directionColumn>
-                <img src={getWeatherSymbol(data.symbol)} alt={data.symbolPhrase} />
-                <p>{capitalize(data.symbolPhrase)}</p>
+                <img src={getWeatherSymbol(data?.symbol)} alt={data?.symbolPhrase} />
+                <p>{capitalize(data?.symbolPhrase)}</p>
               </Flex>
             </S.WidgetImg>
             <S.WidgetDetailsItem color={color}>
-              <span>Cloudiness</span> {data.cloudiness}%
+              <span>Cloudiness</span> {data?.cloudiness}%
             </S.WidgetDetailsItem>
             <S.WidgetDetailsItem color={color}>
-              <span>Precipation</span> {data.precipProb}%, {data.precipRate}&nbsp;mm/h
+              <span>Precipation</span> {data?.precipProb}%, {data?.precipRate}&nbsp;mm/h
```

Comment:
Why you use so many `data?.`
Maybe better will be 
`{data ? <S.Widget color={color}>...</ S.Widget> : <p>some error or something else</p> }`

Reply:
Actually, in the beginning I've had something similar, i.e. `{data && <S.Widget color={color}>...</ S.Widget>}`, but changed to `data?.x` syntax. I don't know if there is much difference here, do you?

Reply:
both ways are valid :)